src/celeritas/ext/detail/G4VecgeomConverter.hh

```diff
@@ -0,0 +1,147 @@
+//----------------------------------*-C++-*----------------------------------//
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+//---------------------------------------------------------------------------//
+/*!
+ * \file G4VecgeomConverter.hh
+ * \brief Class to create a VecGeom model from a pre-existing Geant4 geometry
+ *
+ * Original code from G4VecGeomNav package by John Apostolakis et.al.
+ *
+ * Original source:
+ * https://gitlab.cern.ch/VecGeom/g4vecgeomnav/-/raw/fdd310842fa71c58b3d99646159ef1993a0366b0/include/G4VecGeomConverter.h
+ */
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <utility>
+#include <vector>
+#include <VecGeom/base/TypeMap.h>
+#include <VecGeom/management/GeoManager.h>
+
+#include "Geant4/G4PVReplica.hh"
+#include "Geant4/G4RotationMatrix.hh"
+#include "Geant4/G4ThreeVector.hh"
+
+class G4LogicalVolume;
+class G4AffineTransformation;
+class G4VPhysicalVolume;
+class G4VSolid;
+
+namespace celeritas
+{
+inline namespace VECGEOM_IMPL_NAMESPACE
+{
+class LogicalVolume;
+class Transformation3D;
+class UnplacedBox;
+class VUnplacedVolume;
+class VPlacedVolume;
+
+} // namespace VECGEOM_IMPL_NAMESPACE
+} // namespace vecgeom
+
+// class converting G4 to VecGeom (only geometry; no materials)
+class G4VecGeomConverter
+{
+  private:
+    /** Remember pointer to generated world from imported G4 geometry. */
+    vecgeom::VPlacedVolume const* fWorld;
+
+    // vecgeom::BidirectionalTypeMap<unsigned int, G4VPhysicalVolume const*>
+    // fPlacedVolumeMap; using G4PVReplicaPair = std::pair<G4VPhysicalVolume
+    // const*, int>; vecgeom::BidirectionalTypeMap<unsigned int,
+    // G4PVReplicaPair> fPlacedVolumeMap;
```

Comment:
Is the goal to:
1. for now to "try out" the conversion in Celeritas, then move upstream to VecGeom,
2. maintain our own version (like the BVH navigator) and periodically pull changes from upstream, or
3. keep this implementation for ourselves?

If you think it's choice 3, we should try to conform to our code styles. Otherwise I think the style can be left as-is, but you should still probably delete commented-out code like this.

Reply:
SInce we have decided to add a geometry scaling factor to handle unit conversions, I think it makes sense to move the conversion code into Celeritas -- choice 3 above.
I will be adapting the conversion classes to our code style.